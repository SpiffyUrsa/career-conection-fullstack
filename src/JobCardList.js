import React from "react";
import JobCard from "./JobCard";

/**
 * Renders details about all applicable jobs
 * 
 * Routes --> {CompanyDetail, JobList}-->JobCardList
 * -->JobCard
 * 
 * State
 * 
 * Props
 * 
 */
function JobCardList({ jobs }) {

  const jobsDisplay = jobs.map(job =>
    <JobCard
      id={job.id}
      title={job.title}
      salary={job.salary}
      equity={job.equity}
      key={job.id}
    />)

  return (
    <div>
      {jobsDisplay}
    </div>
  )

}


export default JobCardList;