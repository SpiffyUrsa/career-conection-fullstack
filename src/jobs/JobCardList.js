import React from "react";
import JobCard from "./JobCard";

/**
 * Renders details about all applicable jobs
 * 
 * Routes --> {CompanyDetail, JobList}-->JobCardList
 * -->JobCard
 * 
 * State
 * 
 * Props
 *  - jobs
 */
function JobCardList({ jobs }) {

  const jobsDisplay = !jobs ?
    <h2> No current job openings </h2> : jobs.map(job =>
      <JobCard
        id={job.id}
        title={job.title}
        salary={job.salary}
        equity={job.equity}
        key={job.id}
      />)

  return (
    <div>
      {jobsDisplay}
    </div>
  )

}


export default JobCardList;