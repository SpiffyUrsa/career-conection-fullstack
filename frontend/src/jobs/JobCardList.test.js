import React from "react";
import { render } from "@testing-library/react";
import JobCardList from "./JobCardList";

it("renders without crashing", function() {
  render(<JobCardList />);
});

it("matches snapshot with no jobs", function() {
  const { asFragment } = render(<JobCardList />);
  expect(asFragment()).toMatchSnapshot();
});
