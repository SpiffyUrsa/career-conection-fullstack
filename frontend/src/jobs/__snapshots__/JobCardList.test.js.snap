// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with no jobs 1`] = `
<DocumentFragment>
  <div>
    <h2>
       No current job openings 
    </h2>
  </div>
</DocumentFragment>
`;
